File: training-data/c/count_vowels.c
#include <stdio.h>

int count_vowels(const char *s) {
    int count = 0;
    for (int i = 0; s[i]; i++) {
        char c = s[i];
        if (c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u') count++;
    }
    return count;
}

int main(void) {
    printf("%d\n", count_vowels("hello world"));
    return 0;
}

File: training-data/forth/count_vowels.fs
: VOWEL? ( c -- flag )
  DUP [CHAR] a = SWAP
  DUP [CHAR] e = SWAP
  DUP [CHAR] i = SWAP
  DUP [CHAR] o = SWAP
  [CHAR] u =
  OR OR OR OR ;

: COUNT-VOWELS ( addr len -- count )
  0 -ROT
  OVER + SWAP
  BEGIN
    2DUP <
  WHILE
    DUP C@ VOWEL? IF ROT 1+ ROT SWAP THEN SWAP
    1+
  REPEAT
  2DROP ;

S" hello world" COUNT-VOWELS .
CR
